src/command_modules/azure-cli-resource/azure/cli/command_modules/resource/custom.py

```diff
@@ -122,6 +175,13 @@ def _deploy_arm_template_core(resource_group_name, template_file=None, template_
     else:
         template = get_file_json(template_file)
 
+    missing = _find_missing_parameters(parameters, template)
+    if len(missing) > 0:
+        if prompt_for_parameters:
```

Comment:
We should do the TTY check here and automatically fail if no TTY is detected (see my earlier comment).